linode/manager
# change: [M3-6522] - Tag Component v7 story & cleanup

## Description 📝
This PR replaces the Tag component's `.mdx` story with a fresh v7 story.

## Changes  🔄
- Refactor old mdx story into a v7 story
- Component cleanup
- A new .styles file
- A new unit test
- A few fixes:

**Broken hover state in main menu** 
| Before  | After   |
| ------- | ------- |
| ![Screenshot 2023-10-24 at 16 26 24](https://github.com/linode/manager/assets/130582365/1685e363-f302-48c7-95f8-b785a4746f0f) | ![Screenshot 2023-10-24 at 16 43 36](https://github.com/linode/manager/assets/130582365/b8f93ee0-0721-4ffd-8677-42f9bcef5edc) |

**Poor contrast & accessibility on dark mode**
| Before  | After   |
| ------- | ------- |
| ![Screenshot 2023-10-24 at 16 42 57](https://github.com/linode/manager/assets/130582365/4a9b3851-3d4c-4c4f-bfad-180fde77a402) | ![Screenshot 2023-10-24 at 16 47 22](https://github.com/linode/manager/assets/130582365/c34df465-b246-4eb6-bdf3-c919175cc5d6) |

## Preview 📷
Apart from the fixes shown above, there should be no change in the way the component looks or behaves.

## How to test 🧪

### Prerequisites
- Pull code locally

### Verification steps 
(How to verify changes)
- Navigate to http://localhost:6006/?path=/docs/components-chip-tag--documentation to confirm the new story
- Check the tag in CM to ensure no regression and the small fixes (main search with tagged linodes, tag panels in entity details)

## As an Author I have considered 🤔

*Check all that apply*
